src/endpoint/services.ts

```diff
@@ -0,0 +1,627 @@
+import { Endpoint } from '../endpoint'
+import EndpointClient, { EndpointClientConfig } from '../endpoint-client'
+import { SuccessStatusValue, Status } from '../types'
+
+
+export interface TimeValue {
+	value: string
+}
+
+export interface NumericValue {
+	value: number
+	unit: string
+}
+
+export interface UnitlessNumericValue {
+	value: number
+}
+
+export interface StringValue {
+	value: string
+}
+
+export interface ServiceLocationInfo {
+	locationId: string
+	subscriptions: Array<ServiceLocationInfoSubscriptions>
+	latitude: number
+	longitude: number
+	city: string
+}
+
+export interface ServiceLocationInfoSubscriptions {
+	subscriptionId: string
+	/**
+	 * DIRECT or EXECUTION delivery, default to DIRECT
+	 */
+	type: string
+	/**
+	 * JEXL expression string
+	 */
+	predicate: string
+	subscribedCapabilities: Array<string>
+}
+
+export interface ServiceNewSubscription {
+	/**
+	 * location ID
+	 */
+	locationId: string
+	/**
+	 * subscription ID created
+	 */
+	subscriptionId: string
+}
+
+export enum ServiceCapabilitiesEnum {
+	weather = 'weather',
```

Comment:
Can we chat about why we can't use uppercase for the enum name? When I look at the compiled JavaScript, it does include both parts.

Reply:
What bugs me is the inconsistency between typescript, javascript, and the API itself. If I change the typescript to _WEATHER_ then a javascript client will still need to pass in "weather", and the docs will have to contain both values.

Also, in this particular case "weather" there is an API endpoint that returns the service capabilities available for a location, and the value returned is "weather". So in this particular case we'd probably elect to not have an enum at all, which is probably best now that I think about it.

As for the other enums where the value is not upper case, all options seem nasty to me.